assessments/projects/longhorn/self-assessment.md

```diff
@@ -0,0 +1,436 @@
+| Cryptography                   | Unauthorized data access and tampering       | Enable volume encryption, TLS for data in transit                       | [Encrypting Data at Rest](https://kubernetes.io/docs/tasks/administer-cluster/encrypt-data/) |
+| Multi-tenancy Isolation        | Cross-tenant access or attacks               | Namespace isolation, fine-grained RBAC                                 | [Kubernetes Namespaces](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) |
+| Secrets Management             | Exposure of sensitive data                   | Use Kubernetes Secrets with appropriate access controls                | [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) |
+| Storage                        | Data loss or corruption                      | Replication across nodes, regular snapshots, and backups               | [Longhorn Documentation](https://longhorn.io/docs/) |
+| Authentication                 | Unauthorized system access                   | Strong authentication mechanisms, integration with Kubernetes auth      | [Kubernetes Authentication](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) |
+| Authorization (Access Control) | Unauthorized actions within the storage system| Strict RBAC policies, least privilege access                           | [Kubernetes Authorization](https://kubernetes.io/docs/reference/access-authn-authz/authorization/) |
+| Audit and Logging              | Lack of incident visibility or forensics     | Enable detailed logging, integrate with monitoring tools               | [Kubernetes Logging Architecture](https://kubernetes.io/docs/concepts/cluster-administration/logging/) |
+
+
+### Recommendations
+
+- Regularly update and patch Longhorn and Kubernetes components.
+- Ensure that all communication is encrypted, both in transit and at rest.
+- Conduct periodic access reviews and privilege audits.
+
+
+### Conclusion
+
+- Critical findings should be promptly disclosed to the community.
+- Consideration for additional threat modelling tools, such as attack trees or matrices, to further analyse complex threats.
+</details>
+
+The attact tree for a generic high-level threat scenario is depicted:
+
+<p align="center"><img alt="image" src="https://github.com/Makesh-Srinivasan/ISP_A3/assets/66047630/4f88f24c-14e9-40c1-92e5-ea3761806556"></p>
+
+Each branch of this tree represents a step or method an attacker might use to progress their attack. The end goal is unauthorized access to and exfiltration of sensitive data stored within Longhorn. This attack tree can be used to identify potential vulnerabilities, assess risks, and develop strategies to mitigate or prevent such attacks.
+
+<hr>
+
+## Security functions and features
```

Comment:
These features are meant to aid threat modelling. Hope the threats and recommendations consider them.

Reply:
Dear Ms. Ragashree Shekar, thank you for the comment. Yes, we did indeed use these features in understanding and later building our threat models and analyse the possible threats. Best regards!